### PR TITLE
Cleaner consolidated logic for adding constraints

### LINKED ISSUE #2850: Redundant methods for creating constraints
## Problem

Notice the `db.constraints.operations.create` and `db.constraints.base`, looks like we have 2 ways to achieve creating the same constraint. 

## Expected behavior

We should be having all the constraints creation logic in the `db.constraints.operations.create` and there should be only one method that actually creates different constraints in the db.

## Additional context

Related to RSQLA #2737

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #2850

**Technical details**

This PR removes redundant constraint creation logic from the codebase. To create a foreign key constraint or a unique constraint you can now simply instantiate a `ForeignKeyConstraint` or `UniqueConstraint` respectively with the required parameters and call `add_constraint()` on the created object.

e.g.:
```
django_table.add_constraint(UniqueConstraint(constraint_name, table_oid, [column_attnums]))
```

**OR**

```
uq_constraint = UniqueConstraint(constraint_name, table_oid, [column_attnums])
uq_constraint.add_constraint(engine)
```

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/constraints/operations/create.py [hunk 1]
```diff
@@ -1,34 +1,85 @@
...
+from db.constraints.utils import (
+    get_constraint_match_char_from_type, get_constraint_char_from_action,
+    get_constraint_type_from_char, ConstraintType
+)
 
-def create_constraint(schema, engine, constraint_obj):
-    with engine.begin() as conn:
-        return constraint_obj.add_constraint(schema, engine, conn)
 
+class Constraint():
```
Comment (mathemancer) [inline db/constraints/operations/create.py:9]: What's the point of this class? The methods are both more properly static (though they're set up as instance methods here), and its child classes don't actually inherit anything. 

I think you should remove it.

File: db/constraints/operations/create.py [hunk 2]
```diff
@@ -1,34 +1,85 @@
...
+    def add_constraint(self, engine, table_oid, json_dump):
```
Comment (mathemancer) [inline db/constraints/operations/create.py:10]: This doesn't use `self`.

File: db/constraints/operations/create.py [hunk 3]
```diff
@@ -1,34 +1,85 @@
...
+            engine,
+            'add_constraints',
+            table_oid,
+            json_dump
+        ).fetchone()[0]
 
-def copy_constraint(_, engine, constraint, from_column_attnum, to_column_attnum):
-    constraint_type = get_constraint_type_from_char(constraint.contype)
-    if constraint_type == ConstraintType.UNIQUE.value:
+    def copy_constraint(self, engine, constraint, from_column_attnum, to_column_attnum):
```
Comment (mathemancer) [inline db/constraints/operations/create.py:18]: This doesn't use `self`.

**Review by mathemancer — CHANGES_REQUESTED** [review]
I suggest reorganizing by:
- making the `Constraint` class an abstract base class with a single abstract method: `get_constraint_def_json`. You just want to enforce this interface.
- For the implementations (unique and foreign key):
  - These could (probably should) init with their current signatures.
  - remove the `add_constraint` method and replace it with implementations of `get_constraint_def_json` (this should get the `json.dumps` results).
- Remove the python layer `copy_constraint` logic entirely. It's no longer needed (make sure to merge in develop)
- Add a function `add_constraint` to the module that takes a constraint implementation, gets the json via `conobj.get_constraint_def_json`, and runs the SQL layer `add_constraints` function.

File: db/constraints/operations/create.py (hunk 1 above)
Comment (Anish9901) [reply db/constraints/operations/create.py:9]: Well, I think it should exist for the reason of not repeating the calling of the exact same SQL function from the separate constraint modules.

**Review by mathemancer — APPROVED** [review]
Alright, this is looking good. Thank you!

**Pull request merged**